Automattic/wp-calypso
# chore: cleanup tsconfig for js packages

#### Background

In #53944 we introduced a shared `tsconfig.json` config for all TS packages. This allows us to have a central place to specify `tsc` options and ensure all our packages are consistent.

As JS packages are not compiled or type checked, this shouldn't have any effect on CI or the compilation result. However, having a `tsconfig.json` for JS packages provides a better editor integration and type detection.

VSCode (and I think any other IDE that uses [tsserver](https://github.com/Microsoft/TypeScript/wiki/Standalone-Server-%28tsserver%29)) will try to find the closest `tsconfig.json` when opening a file, assume that's the root of the project and send all the files to `tsserver` for type checking (including JS files). This can be very annoying if you plan to work on a small section of the repo.

#### Changes proposed in this Pull Request

* Adds a shareable `tsconfig.json` for JS packages. 
* Add `tsconfig.json` on packages that were missing it.

#### Testing instructions

Preparation: 
- Open VSCode and open `packages/calypso-doctor/index.js`. Close any other file.
- Restart VSCode by opening the command palette and running "Reload window"
- Mouse over `runEvaluations` and see the popup with the types.

If you do the steps above in `trunk`, the popup will take a few seconds to appear. It should be much faster if you do it in this branch.
